G_channels.md

```diff
@@ -249,6 +249,72 @@ socket = assign(socket, :user, msg["user"])
 
 Sockets store assigned values as a map in `socket.assigns`.
 
+#### Using Token Authentication
+
+When we connect, we'll often need to authenticate the client. Fortunately, this is a fairly easy 3-step process.
+
+Step 1 - Assign Token in Connection:
+
+For example, imagine we have an authentication plug, `MyAuth`, which authenticates the session and assigns a `:current_user`. If the current user exists we can assign the user's token in the connection for use in the layout. To do so, in our "web/router.ex": we'll need to add the `MyAuth` plug and make the assignment, as follows:
+
+```elixir
+pipeline :browser do
+  ...
+  plug MyAuth
+  plug :put_user_token
+end
+
+defp put_user_token(conn, _) do
+  if current_user = conn.assigns[:current_user] do
+    token = Phoenix.Token.sign(conn, "user socket", current_user.id)
+    assign(conn, :user_token, token)
+  else
+    conn
+  end
+end
+```
+
+Step 2 - Pass Token to JS:
+
+Next we need to pass this token to JavaScript. We can do so inside a script tag in `web/templates/layout/app.html.eex`, as follows:
+
+```html
+    <script>window.userToken = "<%%= assigns[:user_token] %>";</script>
+```
+
+We also need to verify the user token in the `connect/2` function in `web/channels/user_socket.ex`, as follows:
+
+```elixir
+def connect(%{"token" => token}, socket) do
+  # max_age: 1209600 is equivalent to two weeks in seconds
+  case Phoenix.Token.verify(socket, "user socket", token, max_age: 1209600) do
+    {:ok, user_id} ->
+      {:ok, assign(socket, :user, user_id)}
+    {:error, reason} ->
+      :error
+  end
+end
+```
+
+Step 3 - Pass Token on Connect:
+
+Finally, we need to pass the token on connect as below, or remove it from connect if we don't care about authentication.
+
+```javascript
+socket.connect()
```

Comment:
we need to show passing the token in the `params` option of the Socket constructor. Otherwise this is set!